src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp

```diff
@@ -293,7 +297,19 @@ class ShenandoahHeap : public CollectedHeap, public ShenandoahSpaceInfo {
 
   size_t _gc_no_progress_count;
 
-  // This updates the singlular, global gc state. This must happen on a safepoint.
+  double*   _gc_historical_utilization;
+  double*   _gc_historical_duration;
+  double    _gcu_historical;
+  size_t    _gc_history_first;
+  intptr_t  _gc_no_progress_count_at_last_oom;
+
+  void capture_gc_no_progress_count_at_last_oom();
+  intptr_t get_gc_no_progress_count_at_last_oom() const;
+
+  void set_gc_state_all_threads(char state);
```

Comment:
This PR looks like it is missing some recent changes, for example: https://github.com/openjdk/jdk/commit/51238c4bdbce48f6b719f7dcfe5b7df8b8c6d85b.